rust/connlib/tunnel/src/gateway.rs

```diff
@@ -67,11 +67,16 @@ where
             ipv6,
             stun(&relays, |addr| self.io.sockets_ref().can_handle(addr)),
             turn(&relays),
-            domain,
-            expires_at,
-            resource,
             Instant::now(),
-        )
+        );
+        let domain_response = self
+            .role_state
+            .allow_access(resource, client_id, expires_at, domain)?;
```

Comment:
@conectado `ClientTunnel::accept` is now a composition of accept + allow access. This makes it obvious that accepting a connection can actually never fail!

I think it would be good to separate this on the client-side. If we'd always first establish the connection and then allow access to it, error handling would be a lot simpler I think because setting up the connection can never fail and we currently don't perform any clean-up for a failed allow-access request.